matrix/matrix.py

```diff
@@ -0,0 +1,193 @@
+from dataclasses import dataclass
+
+from matrix_exceptions import RowTypeNotTuple, RowItemNotNumber, RowAndColumnNumberDifferent, ObjectTypeNotMatrix, \
+                              ObjectTypeNotLegalToMultiply, MatrixNotLegalToMultiply, IllegalInput
+
+# TODO:
+#   Add documentation.
+
+@dataclass(frozen=True)
+class Matrix:
+
+    matrix: tuple = ()
+
+    def __post_init__(self):
+        if self.is_valid_matrix():
+            object.__setattr__(self, "matrix", self.matrix)
+            object.__setattr__(self, "order", len(self.matrix))
+            object.__setattr__(self, "row_pointer", 0)
+
+    def __str__(self):
+        return "%s" % (self.matrix,)
+
+    def __repr__(self):
+        return "Matrix(%s)" % (self.matrix,)
```

Comment:
What happens if we inherit from this class and implement a new one? will we want to get Matrix printed?